src/lazy/encoder/binary/v1_0/value_writer.rs

```diff
@@ -67,7 +64,7 @@ impl<'value, 'top> BinaryValueWriter_1_0<'value, 'top> {
         self.encoding_buffer.as_slice()
     }
 
-    pub fn write_symbol_id(&mut self, symbol_id: SymbolId) -> IonResult<()> {
+    pub fn write_symbol_id(mut self, symbol_id: SymbolId) -> IonResult<()> {
```

Comment:
All of the methods in the `ValueWriter` trait consume `self` to prevent users from misusing it when (e.g.) they go to implement `WriteAsIon`:

```rust
impl WriteAsIon for MyType {
    fn write_as_ion<V: ValueWriter>(&self, writer: V) -> IonResult<()> {
        writer.write_string("foo")?
          .write_i64(1)?
          .write_bool(true)
    }
}
```

This is important for (e.g.) the APIs for writing annotated values and struct fields where there's metadata attached to the single value being produced. (Note that `SequenceWriter` and `StructWriter` implementations take `&mut self` to allow writing many nested values.)